lib/samson/secrets/hashicorp_vault_backend.rb

```diff
@@ -5,20 +5,26 @@ module Secrets
     class HashicorpVaultBackend
       VAULT_SECRET_BACKEND = 'secret/'
       SAMSON_SECRET_NAMESPACE = 'apps/'
+      VAULT_ERR = "Vault backend is down. "
       # we don't really want other directories in the key,
       # and there may be other chars that we find we don't like
       ENCODINGS = {"/": "%2F"}.freeze
 
       class << self
         def read(key)
-          key = vault_path(key)
-          result = vault_client.read(key)
-          return if !result || result.data[:vault].nil?
+          begin
+            key = vault_path(key)
+            result = vault_client.read(key)
+            return if !result || result.data[:vault].nil?
 
-          result = result.to_h
-          result = result.merge(result.delete(:data))
-          result[:value] = result.delete(:vault)
-          result
+            result = result.to_h
+            result = result.merge(result.delete(:data))
+            result[:value] = result.delete(:vault)
+            result
+          rescue Vault::HTTPConnectionError => e
```

Comment:
extract this into a method ... `wrap_vault_errors` ... also raise a specific exception that then can have a nice message / can be rescues in the frontend